File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for ch in s:
            if ch == "{" or ch == "[" or ch == "(" :
                stack.append(ch)
            elif stack and ((ch == "]" and stack[-1] == "[") or (ch == "}" and stack[-1] == "{") or (ch == ")" and stack[-1] == "(")):
                stack.pop()
            else :
                stack.append(ch)
            print(stack)
        if len(stack) == 0:
            return True
        else:
            return False

File: Data Structures & Algorithms/subsets/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    subsets(nums) {
        let results =[];
        let path = [];
        function backtrack(i){
            if(i === nums.length ){
                results.push([...path]);
                return
            }
            path.push(nums[i]);
            backtrack(i+1);
            path.pop();
            backtrack(i+1);


        }
        backtrack(0);
        return results;
         
    }
}
